pkg/executor/insert_common.go

```diff
@@ -1373,7 +1373,11 @@ func (e *InsertValues) removeRow(
 		return true, nil
 	}
 
-	err = r.t.RemoveRecord(e.Ctx().GetTableCtx(), handle, oldRow)
+	if ph, ok := handle.(kv.PartitionHandle); ok {
+		err = e.Table.(table.PartitionedTable).GetPartition(ph.PartitionID).RemoveRecord(e.Ctx().GetTableCtx(), ph.Handle, oldRow)
```

Comment:
Also Note for future reference.
We could fix it by handle partition handle in `func (c *index) GenIndexKey`, but since the `phyTblID` is already included in the index struct, so it is better to add it at the entry of `removeRecord`.